src/components/time-input/TimeInput.tsx

```diff
@@ -57,46 +69,64 @@ const TimeInput = ({
     isRequired = true,
     label,
     onBlur,
-    shouldAutoBlur = true,
+    onChange,
+    shouldAutoFormat = true,
 }: TimeInputProps) => {
-    const [displayTime, setDisplayTime] = React.useState<string | undefined>(
-        initialDate ? intl.formatTime(initialDate) : '',
-    );
+    const [displayTime, setDisplayTime] = React.useState<string>(initialDate ? intl.formatTime(initialDate) : '');
     const [error, setError] = React.useState<React.ReactElement | undefined>(undefined);
 
     /**
      * Handle blur events.
      * Parse and reformat the current display time (as entered by the user).
+     * @param latestValue - string
+     * @param callback - onBlur, onChange, or a combination of the two
+     * @returns
      */
-    const handleBlur = () => {
+    const formatDisplayTime = (latestValue: string = displayTime, callback?: TimeInputEventHandler) => {
+        if (!latestValue) return;
         try {
-            const { hours: parsedHours, minutes: parsedMinutes } = parseTimeFromString(displayTime);
+            const { hours: parsedHours, minutes: parsedMinutes } = parseTimeFromString(latestValue);
             const date = new Date();
             date.setHours(parsedHours);
             date.setMinutes(parsedMinutes);
             const newDisplayTime = intl.formatTime(date);
             setDisplayTime(newDisplayTime);
-            if (onBlur) onBlur({ displayTime: newDisplayTime, hours: parsedHours, minutes: parsedMinutes });
+            if (callback) callback({ displayTime: newDisplayTime, hours: parsedHours, minutes: parsedMinutes });
         } catch (e) {
             setError(<FormattedMessage {...messages.invalidTimeError} />);
         }
     };
 
     /**
-     * Handle keydown events if shouldAutoBlur is true.
-     * If the user pressed Ctrl+Enter or Cmd+Enter (as in a form submit),
-     * blur the input field automatically, which will trigger handleBlur() above.
+     * Handle keydown events if shouldAutoFormat is true.
+     * If the user pressed Ctrl+Enter or Cmd+Enter (as in a form submit), call formatDisplayTime().
      * @param event - KeyboardEvent
      */
     const handleKeyDown = (event: React.KeyboardEvent) => {
         const { ctrlKey, key, metaKey } = event;
         const autoBlurKeyPressed = AUTO_BLUR_KEYS[key] || metaKey || ctrlKey;
-        if (autoBlurKeyPressed) handleBlur();
+        if (autoBlurKeyPressed) {
+            formatDisplayTime(displayTime, parsedData => {
+                if (onBlur) onBlur(parsedData);
+                if (onChange) onChange(parsedData);
+            });
+        }
```

Comment:
I don't think you need the keydown handler with debounced onChange and onBlur implemented as they are now